Core/iDoctor.Application/Services/UserService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using AutoMapper;
+using iDoctor.Application.Dtos.Token;
+using iDoctor.Application.Dtos.UserDtos;
+using iDoctor.Application.Services.Interfaces;
+using iDoctor.Domain.Entities;
+using iDoctor.Domain.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace iDoctor.Application.Services
+{
+    public class UserService:IUserService
+    {
+        private readonly IUserRepository _userRepository;
+        private readonly IPasswordService _passwordService;
+        private readonly ITokenService _tokenService;
+        private readonly IMapper _mapper;
+
+        public UserService(IUserRepository userRepository,
+                           IMapper mapper,
+                           IPasswordService passwordService,
+                           ITokenService tokenService)
+        {
+            _userRepository = userRepository;
+            _mapper = mapper;
+            _passwordService = passwordService;
+            _tokenService = tokenService;
+        }
+
+        public async Task RegisterAsync(RegisterDto model)
+        {
+            model.Password = _passwordService.HashPassword(model.Password);
```

Comment:
It is violation of clean code principles to modify the value of existing property in model. Especially here password doesn't assume hashed password. So I would suggest to map dto to domain model, then assign hashed password to user.HashedPassword = ...